src/main/scala/org/broadinstitute/dsde/workbench/sam/db/tables/PolicyTable.scala

```diff
@@ -20,6 +21,7 @@ object PolicyTable extends SQLSyntaxSupportWithDefaultSamDB[PolicyRecord] {
     rs.get(e.resourceId),
     rs.get(e.groupId),
     rs.get(e.name),
-    rs.get(e.public)
+    rs.get(e.public),
+    rs.longOpt(e.descendantResourceTypeId).map(ResourceTypePK)
```

Comment:
I forgot to update Postgres to actually write this value, but now I'm not sure I even understand this. What if a policy has descendant permissions that apply to multiple different kinds of resource types? Which `ResourceTypePK` would go here?

Reply:
My current thinking is that each role and action in the `PolicyRole` and `PolicyAction` tables has to be associated with a resource type, so there's no need to define which resource type any given policy might apply to using this FK. If there are roles or actions defined in the `PolicyRole` or `PolicyAction` table that descend, then those roles and actions apply to child resources of their given resource type.

Reply:
yes!